intake/source/utils.py

```diff
@@ -245,16 +245,18 @@ def path_to_glob(path):
     if not isinstance(path, str):
         return path
 
-    # Get just the real bits of the urlpath
-    literal_texts = [i[0] for i in fmt.parse(path)]
-
-    # Only use a star for first empty string in literal_texts
-    index_of_empty = [i for i, lt in enumerate(literal_texts) if lt == '' and i != 0]
-    glob = '*'.join([literal_texts[i] for i in range(len(literal_texts)) if i not in index_of_empty])
-
+    # calculate glob expression
+    glob = ''
+    prev_literal = None
+    prev_field_name = None
+    for literal_text, field_name, format_spec, conversion in fmt.parse(path):
+        glob += literal_text
+        if field_name and not('' == literal_text and prev_field_name):  # condition to avoid repeated * on adjacent fields
```

Comment:
This is hard to read. It might be easier without the double negative:

```suggestion
        if field_name and (literal_text or prev_field_name is None):  # condition to avoid repeated * on adjacent fields
```